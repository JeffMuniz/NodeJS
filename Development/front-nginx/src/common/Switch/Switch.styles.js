import styled from "styled-components";

import { white, lighterGreen, ligtherGrey } from "src/styles/colors";

const isLabelBefore = (truly, falsy) => ({ labelBefore }) =>
  labelBefore ? truly : falsy;

export const Wrapper = styled.div`
  width: auto;
  display: flex;
  align-items: center;
`;

export const CheckBox = styled.input`
  position: absolute;
  margin-left: -9999px;
  visibility: hidden;
`;

export const FakeLabel = styled.label`
  order: 2;

  ${CheckBox} + & {
    display: block;
    position: relative;
    cursor: pointer;
    outline: none;
    user-select: none;
    width: 32px;
    height: 16px;
    border-radius: 8px;
  }

  ${CheckBox} + &:before,
  ${CheckBox} + &:after {
    content: "";
    top: 0;
    left: 0;
    bottom: 0;
    display: block;
    position: absolute;
  }

  ${CheckBox} + &:before {
    right: 1px;
    background-color: ${ligtherGrey};
    border-radius: 8px;
    transition: all 0.4s;
  }

  ${CheckBox} + &:after {
    width: 16px;
    background-color: ${white};
    border-radius: 100%;
    box-shadow: 0 1px 1px 1px #D5D5D5;
    transition: all 0.4s;
  }

  ${CheckBox}:checked + &:before {
    background-color: ${lighterGreen};
  }
  ${CheckBox}:checked + &:after {
    transform: translateX(100%);
  }
`;

export const Label = styled.label`
  margin-${isLabelBefore("right", "left")}: 15px;
  order: ${isLabelBefore(1, 2)};
	color: #3F4142;
	font-size: 14px;
	font-weight: bold;
	line-height: 19px;
  letter-spacing: -0.3px;
`;
