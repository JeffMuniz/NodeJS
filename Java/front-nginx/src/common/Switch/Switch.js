import React from "react";
import PropTypes from "prop-types";

import { Label, Wrapper, CheckBox, FakeLabel } from "./Switch.styles";

const Switch = ({
  id,
  labelBefore,
  checked,
  label,
  name,
  onChange,
  ...props
}) => (
  <Wrapper {...props}>
    <CheckBox
      id={id}
      type="checkbox"
      onChange={onChange}
      checked={checked}
      name={name}
    />
    <FakeLabel htmlFor={id} />
    <Label labelBefore={labelBefore}>{label}</Label>
  </Wrapper>
);

Switch.propTypes = {
  id: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  labelBefore: PropTypes.bool,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
};

Switch.defaultProps = {
  checked: false,
  label: "",
  labelBefore: false,
  placeholder: null,
  onChange: () => null,
};

export default Switch;
