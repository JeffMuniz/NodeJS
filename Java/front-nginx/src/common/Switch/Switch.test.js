import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import TestRenderer from "react-test-renderer";
import Switch from "./Switch";

const props = { id: "test", name: "name" };

describe("Switch", () => {
  it("Should render Switch", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<Switch {...props} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("Should return null when call onChange of default props", () => {
    // given
    const testRenderer = TestRenderer.create(<Switch {...props} />);
    const instance = testRenderer.root;

    // when
    const result = instance.props.onChange();

    // then
    expect(result).tomacull();
  });
});
