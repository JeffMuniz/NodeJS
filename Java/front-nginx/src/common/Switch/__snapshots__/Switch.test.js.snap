// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Should render Switch 1`] = `
<styled.div
  placeholder={null}
>
  <styled.input
    checked={false}
    id="test"
    name="name"
    onChange={[Function]}
    type="checkbox"
  />
  <styled.label
    htmlFor="test"
  />
  <styled.label
    labelBefore={false}
  >
    
  </styled.label>
</styled.div>
`;
